### PR TITLE
MM-25292: Fix idempotency regressions

### PR DESCRIPTION
**Pull request by agnivade**
We stop the service first and then copy the binary.
For load agents too, we restart it rather than starting it.

#### Ticket link

https://mattermost.atlassian.net/browse/MM-25292

### CODE REVIEW

File: deployment/terraform/create.go [hunk 1]
```diff
@@ -179,6 +179,12 @@ func (t *Terraform) setupAppServers(output *Output, extAgent *ssh.ExtAgent, uplo
 			// Upload binary if needed.
 			if uploadBinary {
 				mlog.Info("Uploading binary", mlog.String("host", ip))
+				cmd := "sudo service mattermost stop || true"
```
Comment (streamer45) [inline deployment/terraform/create.go:182]: Why do we need the `|| true`? I thought the stop command would succeed whether or not the service is running.

Comment (agnivade) [reply deployment/terraform/create.go:182]: It will fail the first time, because the service isn't even installed.

Comment (streamer45) [reply deployment/terraform/create.go:182]: The service file has been uploaded at that point. Maybe we just need a `systemctl daemon-reload`?

Comment (agnivade) [reply deployment/terraform/create.go:182]: Yes, should be fine.

**Pull request merged**